vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/FileManagerTest.java

```diff
@@ -0,0 +1,143 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import edu.asu.diging.vspace.core.data.FileRepository;
+import edu.asu.diging.vspace.core.exception.FileStorageException;
+import edu.asu.diging.vspace.core.factory.IFileFactory;
+import edu.asu.diging.vspace.core.file.IStorageEngine;
+import edu.asu.diging.vspace.core.model.IVSFile;
+import edu.asu.diging.vspace.core.model.impl.VSFile;
+
+public class FileManagerTest {
+    
+    @Mock
+    private IStorageEngine storageEngine;
+    
+    @Mock
+    private IFileFactory fileFactory;
+    
+    @Mock
+    private FileRepository fileRepo;
+    
+    @InjectMocks
+    private FileManager serviceToTest;
+    
+    private final String fileId = "FILE_ID";
+    private final String fileName = "fileName";
+    private final String fileDescription = "fileName";
+    private final String fileContentString = "file content";
+    
+    
+    @Before
+    public void init() {
+        MockitoAnnotations.initMocks(this);
+    }
+    
+    @Test
+    public void test_getFileById_success() {
+        VSFile file = new VSFile();
+        when(fileRepo.findById(fileId)).thenReturn(Optional.of(file));
+        serviceToTest.getFileById(fileId);
+        verify(fileRepo).findById(fileId);
```

Comment:
we know the method behaves correctly if the returned file has the given id. it really doesn't matter if the repo is called or not. if we exchange the backend or a backend method this test should still pass even if the repo is not called as long as the returned result is correct.